source/mir/random/generic.d

```diff
@@ -0,0 +1,384 @@
+/**
+License: $(LINK2 http://boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors: Sebastian Wilzbach, Ilya Yaroshenko
+
+TODO:
+- automatic method to identify inflection points?
+
+*/
+module mir.random.generic;
+
+alias f_t = double;
+
+import std.stdio;
+
+import mir.random.intern;
+
+/**
+
+
+References:
+    Transformed Density Rejection with Inflection Points)
+
+    Botts, Carsten, Wolfgang Hörmann, and Josef Leydold.
+    "$(LINK2 http://epub.wu-wien.ac.at/3158/1/techreport-110.pdf,
+    Transformed density rejection with inflection points.)"
+    Statistics and Computing 23.2 (2013): 251-260.
+*/
+
+/**
+
+Split the domain of f into three types:
+
+These can be easily found by plotting `f''`.
+$(B Inflection point) can be identified by observing at which points `f''` is 0.
+
+$(UL
+
+    $(LI $(F_TILDE) is entirely concave (`f'' is entirely negative))
+    $(LI $(F_TILDE) is entirely convex (`f'' is entirely positive))
+    $(LI $(F_TILDE) contains one inflection point (`f'' intersects the x-axis once)
+)
+
+Macros:
+    F_TILDE=g
+*/
+
+auto tinflex(T...)(T bs)
+{
+    import std.functional: unaryFun;
+    auto g0 = unaryFun!f0;
+    auto g1 = unaryFun!f1;
+    auto g2 = unaryFun!f2;
+    return tinflexInternal!(g0, g1, g2)(bs);
+}
+
+enum T1Type {T1a, T1b, T2a, T2b, T3a, T3b, T4a, T4b}
```

Comment:
This is very, VERY opaque in its meaning and purpose.  Unless there's a very, very good reason (e.g. matching up with notation in the Tinflex paper) I would try to give both the type and the individual values more semantically meaningful names.

If there is a logic to the naming choices, then it should be documented, because at first glance, it looks arbitrary.